packages/serialize/src/index.ts

```diff
@@ -203,30 +283,39 @@ function handleInterpolation(
         )
         if (matched.length) {
           console.error(
-            '`keyframes` output got interpolated into plain string, please wrap it with `css`.\n\n' +
-              'Instead of doing this:\n\n' +
-              [...matched, `\`${replaced}\``].join('\n') +
-              '\n\nYou should wrap it with `css` like this:\n\n' +
-              `css\`${replaced}\``
+            `\`keyframes\` output got interpolated into plain string, please wrap it with \`css\`.
+
+Instead of doing this:
+
+${[...matched, `\`${replaced}\``].join('\n')}
+
+You should wrap it with \`css\` like this:
+
+css\`${replaced}\``
           )
         }
       }
       break
   }
 
   // finalize string values (regular strings and functions interpolated into css calls)
+  const asString = interpolation as string
   if (registered == null) {
-    return interpolation
+    return asString
   }
-  const cached = registered[interpolation]
-  return cached !== undefined ? cached : interpolation
+  const cached = registered[asString]
+  return cached !== undefined ? cached : asString
 }
 
-function createStringFromObject(
-  mergedProps /*: void | Object */,
-  registered /*: RegisteredCache | void */,
-  obj /*: { [key: string]: Interpolation } */
-) /*: string */ {
+function createStringFromObject<Props>(
```

Comment:
this is an internal function - i wonder if this needs to accept a generic, couldn't we get away with something simpler here? we could try to treat `Props` as `unknown` here, this should still allow us to accept a type parameterized with a generic, while using `ArrayInterpolation<unknown>` (and similar) "internally".

Reply:
Ok!
We don't actually manipulate the `Props` type internally so it should not affect anything